### PR TITLE
Refactoring/16 fix paths

### LINKED ISSUE #16: Replace string value for paths with Path
We use string instead of Path objects in many places in this project. Path objects would be preferable. Replace them where applicable.

### PR DESCRIPTION
**Pull request by MarleneKress79789**
fixes #16

### CODE REVIEW

File: exasol_sphinx_github_pages_generator/cli.py [hunk 1]
```diff
@@ -26,7 +27,7 @@
               help="origin of source_branch. Set to 'tags' "
                    "if your source_branch is a tag")
 @click.option('--source-dir',
-              type=str, default="/doc/",
+              type=str, default="doc/",
```
Comment (tkilias) [inline exasol_sphinx_github_pages_generator/cli.py:30]: click provides also a Path type https://click.palletsprojects.com/en/8.0.x/parameters/

File: exasol_sphinx_github_pages_generator/cli.py [hunk 2]
```diff
@@ -26,7 +27,7 @@
...
               help="Path to the directory inside the source_branch where the "
                    "index.rst and conf.py reside in.")
 @click.option('--module-path',
```
Comment (tkilias) [inline exasol_sphinx_github_pages_generator/cli.py:33]: click provides also a Path type https://click.palletsprojects.com/en/8.0.x/parameters/

File: exasol_sphinx_github_pages_generator/deployer.py [hunk 3]
```diff
@@ -29,7 +29,7 @@ class GithubPagesDeployer:
     def __init__(self, source_dir: str, source_branch: str, source_origin: str, current_commit_id: str,
```
Comment (tkilias) [inline exasol_sphinx_github_pages_generator/deployer.py:29]: Can you sue pathlib.Path also with source_dir?

File: exasol_sphinx_github_pages_generator/deployer.py [hunk 4]
```diff
@@ -128,7 +128,7 @@ def detect_or_verify_source_branch(self) -> None:
             sys.exit(f"Abort, you have uncommitted changes in source branch  {self.source_branch}, "
                      f"please commit and push the following files:\n "
                      f"{uncommitted_changes.stdout}")
-        os.chdir(f".{self.source_dir}")
+        os.chdir(f"./{self.source_dir}")
```
Comment (tkilias) [inline exasol_sphinx_github_pages_generator/deployer.py:131]: Is this relative to current working dir=

File: tests/test_deploy_github_pages.py [hunk 5]
```diff
@@ -25,7 +26,8 @@ def test_remote_branch_creation(setup_test_env):
         "--target-branch", target_branch,
         "--source-branch", source_branch,
         "--module-path", "../test_package"]
-    CliRunner().invoke(cli.main, args_list)
+    result = CliRunner().invoke(cli.main, args_list)
```
Comment (tkilias) [inline tests/test_deploy_github_pages.py:29]: The invoke function has a catch_excecptions option, see https://click.palletsprojects.com/en/7.x/api/#click.testing.CliRunner.invoke 
You can than ask the return value of invoke, for the exception with the exception property

File: noxfile.py [hunk 6]
```diff
@@ -84,7 +84,7 @@ def open_docs(session):
 @nox.session(python=False)
 def tests(session):
     """Run all unit tests"""
-    session.run("pytest", f"{PROJECT_ROOT / 'tests'}")
+    session.run("pytest", "--pdb", f"{PROJECT_ROOT / 'tests'}")
```
Comment (tkilias) [inline noxfile.py:87]: for what do you need --pdb

**Review by tkilias — CHANGES_REQUESTED** [review]

File: noxfile.py (hunk 6 above)
Comment (MarleneKress79789) [reply noxfile.py:87]: i dont, forgott to remove it

File: exasol_sphinx_github_pages_generator/cli.py [hunk 7]
```diff
@@ -92,6 +91,7 @@ def _main():
     except Exception as ex:
         if DEBUG:
             Console.stderr(traceback.format_exc())
+            print(sys.exc_info()[2])
```
Comment (tkilias) [inline exasol_sphinx_github_pages_generator/cli.py:94]: is this a debug output? if yes, please remove, otherwise make more clear what you want to output

**Review by tkilias — CHANGES_REQUESTED** [review]

**Review by tkilias — APPROVED** [review]

**Pull request merged**